### PR TITLE
Implementing OnResponseCompleted

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Related to aspnet/HttpAbstractions#196
@lodejard @Tratcher @GrabYourPitchforks

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@ajaybhargavb__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Ajay Bhargav Baaskaran). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs [hunk 1]
```diff
@@ -303,6 +341,8 @@ public void ProduceEnd(Exception ex)
         {
             ProduceStart();
 
+            FireOnResponseCompleted();
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: Not sure this is in the right place.

Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: Moved.

File: src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs [hunk 2]
```diff
@@ -217,6 +231,30 @@ private void FireOnSendingHeaders()
...
+            {
+                foreach (var entry in onResponseCompleted)
+                {
+                    try
+                    {
+                        entry.Key.Invoke(entry.Value);
+                    }
+                    catch
+                    {
+                        // Ignore exceptions
```
Comment (lodejard) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: We should log a bug to figure out how to get the ILoggerFactory to the server. Doesn't need to block this PR though.

**Comment by lodejard** [comment]
:shipit:

File: src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs (hunk 2 above)
Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: Created an issue. https://github.com/aspnet/KestrelHttpServer/issues/81

Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: Also in Helios. https://github.com/aspnet/Helios/issues/117

**Pull request merged**